# AirBnB NY Locations Data Case Study

In this final project, you task will be to take the data provided and find evidance to answer the following questions.

1. Which hosts are the busiest and why?
2. How many neighborhood groups are available and which shows up the most?
3. Are private rooms the most popular in manhattan?
4. Which hosts are the busiest and based on their reviews?
5. Which neighorhood group has the highest average price?
6. Which neighborhood group has the highest total price?
7. Which top 5 hosts have the highest total price?
8. Who currently has no (zero) availability with a review count of 100 or more?
9. What host has the highest total of prices and where are they located?
10. When did Danielle from Queens last receive a review?

You will be given **4 hours** to complete this assignment. 
**Be Advised** I will go dark for this intire assignment time period. That said, any questions that you would like to ask about the data, or the project **MUST** be asked before the time starts. Once the time has started, I can no longer give information.

This is to similate what you will face when you are out in the wild. 

Happy Coding!

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [3]:
air_bnb = pd.read_csv('./AB_NYC_2019.csv')
air_bnb.head


<bound method NDFrame.head of              id                                               name   host_id  \
0          2539                 Clean & quiet apt home by the park      2787   
1          2595                              Skylit Midtown Castle      2845   
2          3647                THE VILLAGE OF HARLEM....NEW YORK !      4632   
3          3831                    Cozy Entire Floor of Brownstone      4869   
4          5022   Entire Apt: Spacious Studio/Loft by central park      7192   
...         ...                                                ...       ...   
48890  36484665    Charming one bedroom - newly renovated rowhouse   8232441   
48891  36485057      Affordable room in Bushwick/East Williamsburg   6570630   
48892  36485431            Sunny Studio at Historical Neighborhood  23492952   
48893  36485609               43rd St. Time Square-cozy single bed  30985759   
48894  36487245  Trendy duplex in the very heart of Hell's Kitchen  68119814   

         

In [4]:
air_bnb.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   id                              48895 non-null  int64  
 1   name                            48879 non-null  object 
 2   host_id                         48895 non-null  int64  
 3   host_name                       48874 non-null  object 
 4   neighbourhood_group             48895 non-null  object 
 5   neighbourhood                   48895 non-null  object 
 6   latitude                        48895 non-null  float64
 7   longitude                       48895 non-null  float64
 8   room_type                       48895 non-null  object 
 9   price                           48895 non-null  int64  
 10  minimum_nights                  48895 non-null  int64  
 11  number_of_reviews               48895 non-null  int64  
 12  last_review                     

In [5]:
# How many neighborhood groups are available and which shows up the most?

air_bnb.groupby('neighbourhood_group').neighbourhood_group.count().sort_values(ascending=False)



neighbourhood_group
Manhattan        21661
Brooklyn         20104
Queens            5666
Bronx             1091
Staten Island      373
Name: neighbourhood_group, dtype: int64

In [6]:
# Are private rooms the most popular in manhattan?
air_bnb.groupby(['neighbourhood_group','room_type']).neighbourhood_group.count()

#Private rooms are not the most popular. Entire home/apt are the most popular!

neighbourhood_group  room_type      
Bronx                Entire home/apt      379
                     Private room         652
                     Shared room           60
Brooklyn             Entire home/apt     9559
                     Private room       10132
                     Shared room          413
Manhattan            Entire home/apt    13199
                     Private room        7982
                     Shared room          480
Queens               Entire home/apt     2096
                     Private room        3372
                     Shared room          198
Staten Island        Entire home/apt      176
                     Private room         188
                     Shared room            9
Name: neighbourhood_group, dtype: int64

In [15]:
# Which hosts are the busiest and based on their reviews?

air_bnb.groupby('host_name').number_of_reviews.sum().nlargest(10)

host_name
Michael    11081
David       8103
John        7223
Jason       6522
Alex        6204
Chris       5028
Anna        4799
Eric        4733
Daniel      4723
Sarah       4579
Name: number_of_reviews, dtype: int64

In [16]:
air_bnb.groupby('host_name').reviews_per_month.sum().nlargest(10)

host_name
David           508.61
Michael         475.82
Alex            443.44
Sonder (NYC)    397.56
John            321.02
Maria           288.39
Anna            229.99
Andrew          199.35
Jason           196.15
Sam             190.27
Name: reviews_per_month, dtype: float64

In [18]:
#Which neighorhood group has the highest average price?
air_bnb.groupby('neighbourhood_group').price.mean().sort_values(ascending=False)

neighbourhood_group
Manhattan        196.875814
Brooklyn         124.383207
Staten Island    114.812332
Queens            99.517649
Bronx             87.496792
Name: price, dtype: float64

In [21]:
# Which neighborhood group has the highest total price?
air_bnb.groupby('neighbourhood_group').price.max().sort_values(ascending=False)

neighbourhood_group
Queens           10000
Manhattan        10000
Brooklyn         10000
Staten Island     5000
Bronx             2500
Name: price, dtype: int64

In [60]:
#Which top 5 hosts have the highest total price?
air_bnb.groupby('host_name').price.sum().sort_values(ascending=False).nlargest(5)


host_name
Sonder (NYC)    82795
Blueground      70331
Michael         66895
David           65844
Alex            52563
Name: price, dtype: int64

In [37]:
# Who currently has no (zero) availability with a review count of 100 or more?
air_bnb.query("availability_365 == 0 and number_of_reviews >= 100")


id                                       name    host_id  \
8          5203         Cozy Clean Guest Room - Family Apt       7490   
94        20913         Charming 1 bed GR8 WBurg LOCATION!      79402   
132       30031           NYC artists’ loft with roof deck     129352   
174       44221             Financial District Luxury Loft     193722   
180       45556      Fort Greene, Brooklyn: Center Bedroom      67778   
...         ...                                        ...        ...   
29581  22705516         The Quietest Block in Manhattan :)  127740507   
30461  23574142                          queens get away!!  176185168   
31250  24267706  entire sunshine of the spotless mind room   21074914   
32670  25719044                COZY Room for Female Guests   40119874   
35014  27759146     Cozy corner near Empire State Building  209549523   

        host_name neighbourhood_group              neighbourhood  latitude  \
8       MaryEllen           Manhattan            Upper West Side  40.80178   
94     Christiana            Brooklyn               Williamsburg  40.70984   
132           Sol            Brooklyn                 Greenpoint  40.73494   
174         Coral           Manhattan         Financial District  40.70666   
180          Doug            Brooklyn                Fort Greene  40.68863   
...           ...                 ...                        ...       ...   
29581    Kathleen           Manhattan                     Harlem  40.83102   
30461       Janet              Queens                  Laurelton  40.68209   
31250      Albert            Brooklyn         Bedford-Stuyvesant  40.68234   
32670    Stephany            Brooklyn  Prospect-Lefferts Gardens  40.66242   
35014     Mariluz           Manhattan                    Midtown  40.74858   

       longitude        room_type  price  minimum_nights  number_of_reviews  \
8      -73.96723     Private room     79               2                118   
94     -73.95775  Entire home/apt    100               5                168   
132    -73.95030     Private room     50               3                193   
174    -74.01374  Entire home/apt    196               3                114   
180    -73.97691     Private room     65               2                206   
...          ...              ...    ...             ...                ...   
29581  -73.94181     Private room     65               2                103   
30461  -73.73662     Private room     65               1                119   
31250  -73.91318     Private room     49               1                102   
32670  -73.94417     Private room     48               1                131   
35014  -73.98341      Shared room     62               1                112   

      last_review  reviews_per_month  calculated_host_listings_count  \
8      2017-07-21               0.99                               1   
94     2018-07-22               1.57                               1   
132    2019-05-20               1.86                               1   
174    2019-06-20               1.06                               1   
180    2019-06-30               1.92                               2   
...           ...                ...                             ...   
29581  2019-07-07               5.89                               2   
30461  2018-12-24               7.79                               1   
31250  2019-07-05               6.73                               3   
32670  2019-05-31               9.97                               2   
35014  2019-06-13              10.77                               3   

       availability_365  
8                     0  
94                    0  
132                   0  
174                   0  
180                   0  
...                 ...  
29581                 0  
30461                 0  
31250                 0  
32670                 0  
35014                 0  

[162 rows x 16 columns]

In [45]:
# What host has the highest total of prices and where are they located?

air_bnb.groupby(["host_name", "neighbourhood"]).price.sum().nlargest(5)

host_name     neighbourhood     
Sonder (NYC)  Financial District    57738
Red Awning    Midtown               35294
Kevin         Chelsea               20003
Jessica       Clinton Hill          15804
Henry         Upper West Side       15000
Name: price, dtype: int64

In [49]:
# When did Danielle from Queens last receive a review?
air_bnb.query("host_name == 'Danielle' and neighbourhood_group == 'Queens'").last_review.sort_values(ascending=False)


22469    2019-07-08
21517    2019-07-07
22068    2019-07-06
20403    2019-07-06
7086     2019-07-03
33861    2019-06-20
27021    2018-01-02
16349           NaN
Name: last_review, dtype: object

## Further Questions

1. Which host has the most listings?

In [51]:
air_bnb.groupby('host_name').id.count().nlargest(10)

host_name
Michael         417
David           403
Sonder (NYC)    327
John            294
Alex            279
Blueground      232
Sarah           227
Daniel          226
Jessica         205
Maria           204
Name: id, dtype: int64

2. How many listings have completely open availability?

In [53]:
air_bnb.availability_365.value_counts()

0      17533
365     1295
364      491
1        408
89       361
       ...  
195       26
196       24
183       24
181       23
202       20
Name: availability_365, Length: 366, dtype: int64

3. What room_types have the highest review numbers?

In [59]:
air_bnb.groupby('room_type').number_of_reviews.sum()

room_type
Entire home/apt    580403
Private room       538346
Shared room         19256
Name: number_of_reviews, dtype: int64

# Final Conclusion

In this cell, write your final conclusion for each of the questions asked.

Also, if you uncovered some more details that were not asked above, please discribe them here.

-- Add your conclusion --

### 1. There are 5 neighborhood groups with Manhatten showing up the most.
2. Private rooms are not the most popular. Entire Homes/Apartments are the most popular.
3. Micheal is the busiest host based on reviews total while David is the busiest based on reviews per month.
4. Manhatten has the highest average price.
5. Queens, Manhatten, and Brooklyn all have the highest total price.
6. Sonder (NYC), Blueground, Micheal, David, and Alex are the top five with the highest price.
7. There are 162 rows of zero availability.
8. Sonder (NYC), Financial District, $57,738.
9. Danielle received her last review August 7th, 2019.
10. Micheal has the most listings.
11. According to the data, there are 2 possible answers. Either 17533 or 1295 listings are completely available.
12. Entire homes/Apartments have the most reviews.